CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+endif ()
+
+# Specify C++ standard used in the project.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+#[==============================================================================================[
+#                                            Options                                            #
+]==============================================================================================]
+
+include(CMakeDependentOption)
+include(FeatureSummary)
+
+option(TEMPLATE_TESTS "Template: Build test-suite" ${PROJECT_IS_TOP_LEVEL})
+option(TEMPLATE_SHARED_LIBS "Template: Build as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(TEMPLATE_INSTALL "Template: Install project" ${PROJECT_IS_TOP_LEVEL})
+
+
+#[==============================================================================================[
+#                                     Project configuration                                     #
+]==============================================================================================]
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+# Include basic tools
+include(FetchContent)
+if (TEMPLATE_INSTALL)
+	include(CMakePackageConfigHelpers)
+	if (UNIX)
+		include(GNUInstallDirs)
+	endif ()
+endif ()
+
+# Define basic parameters
+if (NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif ()
+set(BUILD_SHARED_LIBS ${TEMPLATE_SHARED_LIBS})
+
+#[==============================================================================================[
+#                                       External packages                                       #
+]==============================================================================================]
+
+FetchContent_Declare(fmt
+		GIT_REPOSITORY https://github.com/fmtlib/fmt
+		GIT_TAG master
+		FIND_PACKAGE_ARGS MODULE
+)
+
+FetchContent_MakeAvailable(fmt)
+
+feature_summary(
+		FILENAME ${CMAKE_CURRENT_BINARY_DIR}/Template.info
+		VAR Template_Info
+		DESCRIPTION "Template features and external libraries"
+		FATAL_ON_MISSING_REQUIRED_PACKAGES
+		WHAT ALL
+)
+
+# Report configuration
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+message(STATUS ${Template_Info})
+
+#[==============================================================================================[
+#                                        Main definition                                        #
+]==============================================================================================]
+
+# Main project
+add_executable(Template_Hello)
+set_target_properties(Template_Hello PROPERTIES
+		EXPORT_NAME Hello
+		OUTPUT_NAME hello
+)
+add_executable(Template::Hello ALIAS Template_Hello)
+add_library(Template_Template)
+set_target_properties(Template_Template PROPERTIES
+		VERSION ${PROJECT_VERSION}
+		SOVERSION ${PROJECT_VERSION_MAJOR}
+		EXPORT_NAME Template
+		OUTPUT_NAME template
+)
+add_library(Template::Template ALIAS Template_Template)
```

Comment:
Another paradigm I would want to see adopted is:
- Namespacing the targets
- Defining the public targets in the top-level CMakeLists